### PR TITLE
Create ResourceCategory model and controller

### PR DESCRIPTION
**Pull request by egiurleo**
This PR accomplishes a few things:
1. Introduces a `ResourceCategory` model with all required fields.
2. Introduces a `ResourceCategories` controller with `show`, `update`, `destroy`, and `create` actions.
3. Adds the following routes to the Rails router:
- GET `/resource-categories/:id` (maps to the `show` action)
- PUT `/resource-categories/:id` (maps to the `update` action)
- POST `/resource-categories/` (maps to the `create` 
- DELETE `/resource-categories/:id` (maps to the `destroy` action)
4. Installs the `rspec-rails` and `factory-bot` gem and uses them to test the `ResourceCategories` router
5. Updates the `CONTRIBUTING.md` file with this repo's name and more information about testing.

You can now run tests in this repository with the command `bundle exec rspec`.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
+require "#{Rails.root}/lib/helpers/api_helpers"
```
Comment (egiurleo) [inline app/controllers/application_controller.rb]: Does anyone have a preference between explicitly requiring files in `lib` vs. autoloading them by default?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
...
+
 class ApplicationController < ActionController::API
+  include APIHelpers
```
Comment (egiurleo) [inline app/controllers/application_controller.rb]: My intention here is to have some helper methods that are available to all the controllers (in this case, it's a method that generates a list of acceptable upsert parameters based on the model). I know there are a few Rails-y patterns to accomplish this goal, so I'm open to feedback here.

File: app/controllers/resource_categories_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+class ResourceCategoriesController < ApplicationController
+  before_action :find_resource_category, only: [:show, :update, :destroy]
+
+  def show
+    render json: @resource_category
+  end
+
+  def create
+    # TODO: authenticate admin user
```
Comment (egiurleo) [inline app/controllers/resource_categories_controller.rb]: Only admin users in the CMS are going to have the ability to create, update, and destroy, but we don't have a user model or authentication method yet -- coming in a follow-up PR.

File: db/migrate/20190715133218_create_resource_categories.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+class CreateResourceCategories < ActiveRecord::Migration[5.2]
+  def change
+    create_table :resource_categories do |t|
+      t.string  :name
+      t.text    :short_description
+      t.text    :description
+      t.binary  :icon
```
Comment (egiurleo) [inline db/migrate/20190715133218_create_resource_categories.rb]: I believe the idea was to store svg images directly in the database. I don't have a ton of experience doing that (my tendency is to store images in an S3 bucket or something and then just store a reference to them), but I read that it's doable if you store a binary string. Open to discussion and guidance here.

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
+# This file is copied to spec/ when you run 'rails generate rspec:install'
```
Comment (egiurleo) [inline spec/rails_helper.rb]: Automatically generated by `rspec-rails`.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (JZenk) [reply app/controllers/application_controller.rb]: For me it depends on the size of the file, how many files use it, and the load time.  I am also curious if anyone else has opinions.

**Pull request merged**